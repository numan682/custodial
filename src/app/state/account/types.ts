import { Transaction } from 'app/state/transaction/types'
import { ErrorPayload } from 'types/errors'
import { StringifiedBigInt } from 'types/StringifiedBigInt'

export interface BalanceDetails {
  available: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  debonding: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  delegations: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  total: StringifiedBigInt | null
}

export interface Allowance {
  address: string
  amount: StringifiedBigInt
}

export interface Account extends BalanceDetails {
  address: string
  allowances?: Allowance[]
}

/* --- STATE --- */
export interface AccountState extends Account {
  loading: boolean
  accountError?: ErrorPayload
  transactions: Transaction[]
  transactionsError?: ErrorPayload
}
