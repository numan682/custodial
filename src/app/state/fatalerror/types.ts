/* --- STATE --- */
export interface FatalErrorPayload {
  message: string
  stack?: string
  sagaStack?: string
}

export interface FatalErrorState {
  error?: FatalErrorPayload
}
