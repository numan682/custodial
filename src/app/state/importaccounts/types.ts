import { ErrorPayload } from 'types/errors'
import { WalletType } from '../wallet/types'
import { BalanceDetails } from '../account/types'

/* --- STATE --- */
export interface ImportAccountsListAccount {
  address: string
  balance?: BalanceDetails
  path: number[]
  pathDisplay: string
  privateKey?: string
  publicKey: string
  selected: boolean
  type: WalletType
}

export enum ImportAccountsStep {
  Idle = 'idle',
  OpeningUSB = 'opening_usb',
  LoadingAccounts = 'loading_accounts',
  LoadingBalances = 'loading_balances',
}

export interface ImportAccountsState {
  accounts: ImportAccountsListAccount[]
  error?: ErrorPayload
  showAccountsSelectionModal: boolean
  accountsSelectionPageNumber: number
  step: ImportAccountsStep
}
