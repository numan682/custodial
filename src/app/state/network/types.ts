/* --- STATE --- */

export type NetworkType = 'local' | 'testnet' | 'mainnet'

export interface NetworkState {
  selectedNetwork: NetworkType

  /** Ticker symbol (ROSE, TEST, etc.) */
  ticker: string

  /** chainContext / Genesis Hash */
  chainContext: string

  /** Current epoch */
  epoch: number

  /** Minimum staking amount */
  minimumStakingAmount: number
}
