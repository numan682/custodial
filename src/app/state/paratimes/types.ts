import { StringifiedBigInt } from 'types/StringifiedBigInt'
import { ErrorPayload } from 'types/errors'
import { ParaTime } from '../../../config'

export enum TransactionFormSteps {
  TransferType = 'transferType',
  ParaTimeSelection = 'paraTimeSelection',
  TransactionRecipient = 'transactionRecipient',
  TransactionAmount = 'transactionAmount',
  TransactionConfirmation = 'transactionConfirmation',
  TransactionSummary = 'transactionSummary',
  TransactionError = 'transactionError',
}

export enum TransactionTypes {
  Deposit = 'deposit',
  Withdraw = 'withdraw',
}

export interface TransactionForm {
  amount: string
  confirmTransfer: boolean
  confirmTransferToValidator: boolean
  confirmTransferToForeignAccount: boolean
  defaultFeeAmount: string
  ethPrivateKey: string
  feeAmount: string
  feeGas: string
  paraTime?: ParaTime
  recipient: string
  type: TransactionTypes | undefined
}

export interface ParaTimesState {
  balance: StringifiedBigInt
  isLoading: boolean
  transactionError?: ErrorPayload
  transactionForm: TransactionForm
  transactionFormStep: TransactionFormSteps
}

export type Runtime = {
  address: string
  id: string
  decimals: number
  feeGas: bigint
}

export type ParaTimeTransaction = Pick<
  TransactionForm,
  'amount' | 'ethPrivateKey' | 'feeAmount' | 'feeGas' | 'recipient' | 'type'
>
