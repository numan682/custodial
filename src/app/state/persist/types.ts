import { RootState } from 'types'
// eslint-disable-next-line @typescript-eslint/ban-types
interface PreserveAliasName extends String {}

export interface KeyWithSalt {
  key: Uint8Array
  salt: Uint8Array
}
/** Redux can't serialize Uint8Arrays so we base64 encode and stringify */
export type StringifiedKeyWithSalt = string & PreserveAliasName

export interface EncryptedObject {
  secretbox: Uint8Array
  nonce: Uint8Array
  salt: Uint8Array
}
/** Redux can't serialize Uint8Arrays so we base64 encode and stringify */
export type EncryptedString = string & PreserveAliasName
// Hover to check if inferred variable type is EncryptedString (not string)
export const testPreserveAliasName = '{}' as EncryptedString

export interface SetUnlockedRootStatePayload {
  persistedRootState: PersistedRootState
  stringifiedEncryptionKey: StringifiedKeyWithSalt
}

/* --- STATE --- */
export interface PersistState {
  hasPersistedProfiles: boolean
  isPersistenceUnsupported: boolean
  loading: boolean
  stringifiedEncryptionKey: undefined | 'skipped' | StringifiedKeyWithSalt
  enteredWrongPassword: boolean
}

export interface PersistedRootState extends Pick<RootState, 'theme' | 'wallet' | 'network'> {}
export interface SyncedRootState extends Pick<RootState, 'theme' | 'wallet' | 'network' | 'persist'> {}
