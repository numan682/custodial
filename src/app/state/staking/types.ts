/* --- STATE --- */

import { NetworkType } from 'app/state/network/types'
import { ErrorPayload } from 'types/errors'
import { StringifiedBigInt } from 'types/StringifiedBigInt'

export interface Validator {
  address: string
  name?: string
  nodeAddress: string
  escrow: StringifiedBigInt
  status: 'active' | 'inactive' | 'unknown'
  rank: number
  media?: ValidatorMediaInfo
  current_rate?: number
  current_commission_bound?: CommissionBound
}

export interface Validators {
  timestamp: number
  network: NetworkType
  list: Validator[]
}

export interface CommissionBound {
  lower: number
  upper: number
  epochStart: number
  epochEnd?: number
}

export interface ValidatorDetails {
  currentCommissionBounds?: CommissionBound
  scheduledCommissionBounds?: CommissionBound[]
  nextCommission?: {
    fee: number
    epochStart: number
  }
}
export interface ValidatorMediaInfo {
  website_link?: string
  email_address?: string
  twitter_acc?: string
  tg_chat?: string
  logotype?: string
}

export interface Delegation {
  validator?: Validator
  amount: StringifiedBigInt
  shares: StringifiedBigInt
  validatorAddress: string
}

export interface DebondingDelegation extends Delegation {
  epoch: number
}

export interface StakingState {
  /** List of all the validators */
  validators: Validators | null

  /** Error from last attempt to update our list of validators */
  updateValidatorsError?: ErrorPayload

  /** List of active delegations for the selected account */
  delegations: Delegation[] | null

  /** List of debonding delegations for the selected account */
  debondingDelegations: DebondingDelegation[] | null

  /** Error from fetching delegations */
  updateDelegationsError?: ErrorPayload

  /** Address of the selected validator */
  selectedValidator?: string | null

  /** Lazy-loaded validator details (commission bounds and rates schedules) */
  selectedValidatorDetails: ValidatorDetails | null

  loading: boolean
}
