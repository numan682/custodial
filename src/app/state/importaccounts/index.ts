import { PayloadAction } from '@reduxjs/toolkit'
import { ErrorPayload } from 'types/errors'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { ImportAccountsListAccount, ImportAccountsState, ImportAccountsStep } from './types'

export const initialState: ImportAccountsState = {
  accounts: [],
  showAccountsSelectionModal: false,
  accountsSelectionPageNumber: 0,
  step: ImportAccountsStep.Idle,
}

const slice = createSlice({
  name: 'importAccounts',
  initialState,
  reducers: {
    clear(state) {
      state.accounts = []
      state.error = undefined
      state.step = ImportAccountsStep.Idle
      state.showAccountsSelectionModal = false
    },
    enumerateAccountsFromLedger(state) {
      state.accounts = []
      state.accountsSelectionPageNumber = 0
      state.showAccountsSelectionModal = true
      state.step = ImportAccountsStep.Idle
    },
    enumerateMoreAccountsFromLedger(state) {
      state.step = ImportAccountsStep.Idle
    },
    enumerateAccountsFromMnemonic(state, _action: PayloadAction<string>) {
      state.step = ImportAccountsStep.Idle
      state.accounts = []
      state.accountsSelectionPageNumber = 0
      state.showAccountsSelectionModal = true
    },
    toggleAccount(state, action: PayloadAction<string>) {
      const account = state.accounts.find(a => a.address === action.payload)!
      account.selected = !account.selected
    },
    accountGenerated(state, action: PayloadAction<ImportAccountsListAccount>) {
      state.accounts.push(action.payload)
    },
    accountsListed(state, action: PayloadAction<ImportAccountsListAccount[]>) {
      state.accounts = action.payload
    },
    updateAccountBalance(
      state,
      action: PayloadAction<Pick<ImportAccountsListAccount, 'address' | 'balance'>>,
    ) {
      const { address, balance } = action.payload
      const account = state.accounts.find(a => a.address === address)
      if (account) account.balance = balance
    },
    setPage(state, action: PayloadAction<number>) {
      state.accountsSelectionPageNumber = action.payload
    },
    setStep(state, action: PayloadAction<ImportAccountsStep>) {
      state.step = action.payload
    },
    operationFailed(state, action: PayloadAction<ErrorPayload>) {
      state.error = action.payload
      state.step = ImportAccountsStep.Idle
    },
  },
})

export const { actions: importAccountsActions } = slice

export default slice.reducer
