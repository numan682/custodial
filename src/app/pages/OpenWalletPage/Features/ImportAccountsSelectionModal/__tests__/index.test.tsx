import { act, cleanup, render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { importAccountsActions } from 'app/state/importaccounts'
import { ImportAccountsStep } from 'app/state/importaccounts/types'
import { walletActions } from 'app/state/wallet'
import * as React from 'react'
import { Provider, useDispatch } from 'react-redux'
import { configureAppStore } from 'store/configureStore'
import { ThemeProvider } from 'styles/theme/ThemeProvider'
import { WalletType } from 'app/state/wallet/types'
import { ImportAccountsSelectionModal } from '..'

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}))

const renderComponent = (store: any, abortFunction = () => {}) =>
  render(
    <Provider store={store}>
      <ThemeProvider>
        <ImportAccountsSelectionModal abort={abortFunction} type={WalletType.Ledger} />
      </ThemeProvider>
    </Provider>,
  )

describe('<ImportAccountsSelectionModal  />', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    store = configureAppStore()
    jest.mocked(useDispatch).mockImplementation(() => jest.fn())
  })

  afterEach(() => {
    jest.resetModules()
    jest.resetAllMocks()
    cleanup()
    document.body.innerHTML = ''
  })

  it('should match snapshot', () => {
    const component = renderComponent(store)
    expect(component.baseElement).toMatchSnapshot()
    screen.getByTestId('ledger-open-accounts')
  })

  it('should list the accounts when done', () => {
    const component = renderComponent(store)
    act(() => {
      store.dispatch(
        importAccountsActions.accountsListed([
          {
            address: 'oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe',
            balance: { available: '0', debonding: '0', delegations: '0', total: '0' },
            path: [44, 474, 0],
            pathDisplay: `m/44'/474'/0'`,
            publicKey: '00',
            selected: false,
            type: WalletType.Mnemonic,
          },
        ]),
      )

      store.dispatch(importAccountsActions.setStep(ImportAccountsStep.Idle))
    })
    expect(component.getByText('oasis1qzyq...7gh2pxqe')).toBeInTheDocument()
  })

  it('should open the selected accounts', async () => {
    const dispatchFn = jest.fn()
    jest.mocked(useDispatch).mockImplementation(() => dispatchFn)

    renderComponent(store)
    act(() => {
      store.dispatch(
        importAccountsActions.accountsListed([
          {
            address: 'oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe',
            balance: { available: '0', debonding: '0', delegations: '0', total: '0' },
            path: [44, 474, 0],
            pathDisplay: `m/44'/474'/0'`,
            publicKey: '00',
            selected: false,
            type: WalletType.Mnemonic,
          },
          {
            address: 'oasis1qqv25adrld8jjquzxzg769689lgf9jxvwgjs8tha',
            balance: { available: '0', debonding: '0', delegations: '0', total: '0' },
            path: [44, 474, 1],
            pathDisplay: `m/44'/474'/1'`,
            publicKey: '00',
            selected: false,
            type: WalletType.Mnemonic,
          },
        ]),
      )

      store.dispatch(importAccountsActions.setStep(ImportAccountsStep.Idle))
    })
    await userEvent.click(screen.getByText('oasis1qzyq...7gh2pxqe'))
    expect(dispatchFn).toHaveBeenLastCalledWith({
      payload: 'oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe',
      type: importAccountsActions.toggleAccount.type,
    })
    act(() => {
      store.dispatch(importAccountsActions.toggleAccount('oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe'))
    })
    await userEvent.click(screen.getByTestId('ledger-open-accounts'))
    expect(dispatchFn).toHaveBeenLastCalledWith(
      expect.objectContaining({ type: walletActions.openWalletsFromLedger.type }),
    )
  })
})
