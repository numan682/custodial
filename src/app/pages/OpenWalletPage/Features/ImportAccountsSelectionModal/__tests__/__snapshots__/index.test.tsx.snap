// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImportAccountsSelectionModal  /> should match snapshot 1`] = `
.c0 {
  font-family: Rubik;
  font-size: 18px;
  line-height: 24px;
  background-color: #ffffff;
  color: #16213E;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 800px;
  padding: 24px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-top: 24px;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
}

.c8 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 24px;
}

.c13 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 48px;
}

.c17 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 12px;
}

.c9 {
  font-size: 18px;
  line-height: 24px;
  font-weight: normal;
}

.c6 {
  margin-top: 0px;
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c1 {
  font-family: Rubik;
  font-size: 18px;
  line-height: 24px;
  background-color: #ffffff;
  color: #16213E;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  background: transparent;
  position: relative;
  z-index: 20;
  pointer-events: none;
  outline: none;
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
}

.c2 {
  position: absolute;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  background-color: rgba(0,0,0,0.5);
  color: #F8F8F8;
  pointer-events: all;
  will-change: transform;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 48px;
  background-color: #FFFFFF;
  color: #16213E;
  outline: none;
  pointer-events: all;
  z-index: 20;
  position: absolute;
  max-height: calc(100% - 0px - 0px);
  max-width: calc(100% - 0px - 0px);
  border-radius: 4px;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  -webkit-animation: hWAzxx 0.2s ease-in-out forwards;
  animation: hWAzxx 0.2s ease-in-out forwards;
}

.c4 {
  width: 0;
  height: 0;
  overflow: hidden;
  position: absolute;
}

.c12 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #0092f6;
  border-radius: 18px;
  color: #16213E;
  padding: 4px 20px;
  font-size: 14px;
  line-height: 20px;
  opacity: 0.3;
  cursor: default;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  font-weight: bold;
}

.c12:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c12:focus > circle,
.c12:focus > ellipse,
.c12:focus > line,
.c12:focus > path,
.c12:focus > polygon,
.c12:focus > polyline,
.c12:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c12:focus::-moz-focus-inner {
  border: 0;
}

.c12:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c12:focus:not(:focus-visible) > circle,.c12:focus:not(:focus-visible) > ellipse,
.c12:focus:not(:focus-visible) > line,.c12:focus:not(:focus-visible) > path,
.c12:focus:not(:focus-visible) > polygon,.c12:focus:not(:focus-visible) > polyline,
.c12:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c12:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c14 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #0092f6;
  border-radius: 18px;
  color: #16213E;
  padding: 4px 20px;
  font-size: 14px;
  line-height: 20px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  font-weight: bold;
}

.c14:hover {
  box-shadow: 0px 0px 0px 2px #0092f6;
}

.c14:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c14:focus > circle,
.c14:focus > ellipse,
.c14:focus > line,
.c14:focus > path,
.c14:focus > polygon,
.c14:focus > polyline,
.c14:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c14:focus::-moz-focus-inner {
  border: 0;
}

.c14:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c14:focus:not(:focus-visible) > circle,.c14:focus:not(:focus-visible) > ellipse,
.c14:focus:not(:focus-visible) > line,.c14:focus:not(:focus-visible) > path,
.c14:focus:not(:focus-visible) > polygon,.c14:focus:not(:focus-visible) > polyline,
.c14:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c14:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c16 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #0092f6;
  border-radius: 4px;
  color: #16213E;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  font-weight: bold;
}

.c16:hover {
  box-shadow: 0px 0px 0px 2px #0092f6;
}

.c16:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c16:focus > circle,
.c16:focus > ellipse,
.c16:focus > line,
.c16:focus > path,
.c16:focus > polygon,
.c16:focus > polyline,
.c16:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c16:focus::-moz-focus-inner {
  border: 0;
}

.c16:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c16:focus:not(:focus-visible) > circle,.c16:focus:not(:focus-visible) > ellipse,
.c16:focus:not(:focus-visible) > line,.c16:focus:not(:focus-visible) > path,
.c16:focus:not(:focus-visible) > polygon,.c16:focus:not(:focus-visible) > polyline,
.c16:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c16:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c18 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #0092f6;
  border-radius: 4px;
  color: #16213E;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #0092f6;
  color: #F8F8F8;
  border-radius: 4px;
  opacity: 0.3;
  cursor: default;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  font-weight: bold;
}

.c18:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c18:focus > circle,
.c18:focus > ellipse,
.c18:focus > line,
.c18:focus > path,
.c18:focus > polygon,
.c18:focus > polyline,
.c18:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c18:focus::-moz-focus-inner {
  border: 0;
}

.c18:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c18:focus:not(:focus-visible) > circle,.c18:focus:not(:focus-visible) > ellipse,
.c18:focus:not(:focus-visible) > line,.c18:focus:not(:focus-visible) > path,
.c18:focus:not(:focus-visible) > polygon,.c18:focus:not(:focus-visible) > polyline,
.c18:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c18:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

@media only screen and (max-width:768px) {
  .c5 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-flex-basis: auto;
    -ms-flex-preferred-size: auto;
    flex-basis: auto;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    padding-top: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    padding-top: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    width: auto;
    height: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c13 {
    width: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c17 {
    width: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    margin-top: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    position: fixed;
    width: 100%;
    height: 100%;
    min-height: 100vh;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    position: relative;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    position: relative;
    max-height: none;
    max-width: none;
    border-radius: 0;
    height: 100vh;
    width: 100vw;
  }
}

<body
  style=""
>
  <div>
    <div
      class="c0"
      style="min-height: 100vh;"
    />
  </div>
  <div>
    <div
      aria-hidden="false"
    >
      <div
        class="c1"
        tabindex="-1"
      >
        <div
          class="c2"
        />
        <div
          class="c3"
          data-g-portal-id="0"
          style="overflow-y: auto;"
        >
          <a
            aria-hidden="true"
            class="c4"
            tabindex="-1"
          />
          <div>
            <form
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              spellcheck="false"
            >
              <div
                class="c5"
              >
                <h2
                  class="c6"
                >
                  <div
                    class="c7"
                  >
                    openWallet.importAccounts.selectWallets
                    <div
                      class="c8"
                    />
                    <span
                      class="c9"
                    >
                      openWallet.importAccounts.accountCounter
                    </span>
                  </div>
                </h2>
                <div
                  class="c10"
                  style="min-height: 362px;"
                >
                  <div
                    class="c10"
                  />
                </div>
                <div
                  class="c11"
                >
                  <button
                    aria-label="openWallet.importAccounts.prev"
                    class="c12"
                    disabled=""
                    type="button"
                  >
                    &lt;
                  </button>
                  <div
                    class="c13"
                  />
                  <div
                    class="c10"
                  >
                    <span>
                      openWallet.importAccounts.pageNumber
                    </span>
                  </div>
                  <div
                    class="c13"
                  />
                  <button
                    aria-label="openWallet.importAccounts.next"
                    class="c14"
                    type="button"
                  >
                    &gt;
                  </button>
                </div>
                <p>
                  persist.createProfile.unsupported
                </p>
                <div
                  class="c15"
                >
                  <button
                    class="c16"
                    type="button"
                  >
                    common.cancel
                  </button>
                  <div
                    class="c17"
                  />
                  <button
                    class="c18"
                    data-testid="ledger-open-accounts"
                    disabled=""
                    type="submit"
                  >
                    openWallet.importAccounts.openWallets
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
